bus-mapping/src/evm/opcodes/begin_end_tx.rs

```diff
@@ -0,0 +1,334 @@
+use super::TxExecSteps;
+use crate::{
+    circuit_input_builder::{CircuitInputStateRef, ExecState, ExecStep},
+    operation::{AccountField, AccountOp, CallContextField, TxReceiptField, TxRefundOp, RW},
+    state_db::CodeDB,
+    Error,
+};
+use eth_types::{
+    evm_types::{GasCost, MAX_REFUND_QUOTIENT_OF_GAS_USED},
+    evm_unimplemented, ToWord, Word,
+};
+use ethers_core::utils::get_contract_address;
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct BeginEndTx;
+
+impl TxExecSteps for BeginEndTx {
+    fn gen_associated_steps(
+        state: &mut CircuitInputStateRef,
+        execution_step: ExecState,
+    ) -> Result<ExecStep, Error> {
+        match execution_step {
+            ExecState::BeginTx => gen_begin_tx_steps(state),
+            ExecState::EndTx => gen_end_tx_steps(state),
+            _ => {
+                evm_unimplemented!("unsupported tx execution step {:?}", execution_step);
+                Ok(state.new_begin_tx_step())
```

Comment:
Same as above

Reply:
fixed in fixed in [b737885](https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1336/commits/b737885318cb7fc714406a0735502980d40db5cb)